typescript-async-await/exceptions.ts

```diff
@@ -0,0 +1,46 @@
+import { read } from './read.js';
+
+// Keep track of app load time so each log message can be timed.
+// Log messages should all be approximately 1 second apart.
+const startTime = Date.now();
+const elapsed = (): string =>
+  `${Math.round((Date.now() - startTime) / 1000)}s -`;
+
+async function throwOnce(): Promise<void> {
+  try {
+    const msg = await read('foo', false);
+    console.log(elapsed(), 'throwOnce:', msg);
+  } catch (error: any) {
+    console.log(elapsed(), 'throwOnce Error:', error);
+  }
+}
+
+async function throwSeveral(): Promise<void> {
+  try {
+    const msg = await read('foo1', false);
+    console.log(elapsed(), 'throwSeveral1:', msg);
+    const msg1 = await read('foo2', false);
+    console.log(elapsed(), 'throwSeveral2:', msg1);
+    const msg2 = await read('foo3', false);
+    console.log(elapsed(), 'throwSeveral3:', msg2);
+  } catch (error: any) {
+    console.log(elapsed(), 'throwSeveral Error:', error);
+  }
+}
+
+async function throwChained(): Promise<void> {
+  try {
+    const msg1 = await read('foo-chain', false);
+    console.log(elapsed(), 'throwChained1:', msg1);
+    const msg2 = await read(msg1, false);
+    console.log(elapsed(), 'throwChained2:', msg2);
+    const msg3 = await read(msg2, false);
+    console.log(elapsed(), 'throwChained3:', msg3);
+  } catch (error: any) {
+    console.log(elapsed(), 'throwChained Error:', error);
+  }
+}
+
+throwOnce()
+  .then(() => throwSeveral())
+  .then(() => throwChained());
```

Comment:
Same thing here, if your `throwOnce` doesn't get fulfilled, you won't run the other functions. You should _await_ each call.